Support Alpine builds

Hi!

We tried to package arkime in an alpine docker image, but the build fails with:

```sh
/usr/lib/gcc/x86_64-alpine-linux-musl/13.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reader-tpacketv3.o: in function `reader_tpacketv3_init':
/src/capture/reader-tpacketv3.c:239:(.text+0x972): undefined reference to `mmap64'
collect2: error: ld returned 1 exit status
```

A quick google showed that musl does not use mmap64 by design and they have a [strong opinion](https://wiki.musl-libc.org/faq.html#Q:-Do-I-need-to-define-%3Ccode%3E_LARGEFILE64_SOURCE%3C/code%3E-to-get-64bit-%3Ccode%3Eoff_t%3C/code%3E?) about it:

> Q: Do I need to define _LARGEFILE64_SOURCE to get 64bit off_t?
>
> ...
>
> -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE have nothing to do with supporting large files properly but with exposing the idiotic legacy interfaces with 64 on the end of their names (like open64, lseek64, etc.). only -D_FILE_OFFSET_BITS=64 should ever be used, anywhere, even on glibc.

I have no idea if that is true or not, but this patch makes arkime compile under alpine.

> When the sources are translated with _FILE_OFFSET_BITS == 64 this function is actually available under the name mmap. I.e., the new, extended API using 64 bit file sizes and offsets transparently replaces the old API.

```patch
---
 capture/reader-tpacketv3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/capture/reader-tpacketv3.c b/capture/reader-tpacketv3.c
index 6d4eacc..14a00be 100644
--- a/capture/reader-tpacketv3.c
+++ b/capture/reader-tpacketv3.c
@@ -235,7 +235,7 @@ void reader_tpacketv3_init(char *UNUSED(name))
                     CONFIGEXIT("Error setting SO_ATTACH_FILTER: %s", strerror(errno));
             }
 
-            infos[i][t].map = mmap64(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
+            infos[i][t].map = mmap(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
                                      PROT_READ | PROT_WRITE, MAP_SHARED | MAP_LOCKED, infos[i][t].fd, 0);
             if (unlikely(infos[i][t].map == MAP_FAILED)) {
                 CONFIGEXIT("MMap64 failure in reader_tpacketv3_init, %d: %s. Tried to allocate %d bytes (tpacketv3BlockSize: %d * 64) which was probbaly too large for this host, you probably need to reduce one of the values.", errno, strerror(errno), infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr, blocksize);
-- 
2.25.1
```

And from what I could gather, it seems to have no implications, as [mmap is always 64 bit](https://www.gnu.org/software/libc/manual/html_node/Memory_002dmapped-I_002fO.html#index-mmap64).

We used the following Dockerfile:

```Dockerfile
ARG ARKIME_VERSION=v5.3.0
ARG ARKIME_ARCHIVE=refs/tags/${ARKIME_VERSION}

FROM alpine:3.19.0 as alpine-base

FROM alpine-base as arkime-src
ARG ARKIME_VERSION
ARG ARKIME_ARCHIVE

WORKDIR /src
RUN wget -q https://github.com/arkime/arkime/archive/$ARKIME_ARCHIVE.tar.gz -O - > /tmp/arkime-$ARKIME_VERSION.tar.gz
RUN tar --strip-components=1 -xzf /tmp/arkime-$ARKIME_VERSION.tar.gz --directory=/src/

FROM alpine-base as compile

WORKDIR /src

COPY --from=arkime-src /src /src

RUN mkdir /opt/arkime

RUN apk add autoconf automake

RUN autoupdate
RUN autoreconf --force --install

# configure: error: no acceptable C compiler found in $PATH
# same as binutils file fortify-headers g++ gcc make musl-dev patch
RUN apk add build-base
# configure: error: please install PCRE library
RUN apk add pcre-dev
# configure: error: please install uuid library
RUN apk add util-linux-dev
# configure: error: libpcap not found
RUN apk add libpcap-dev
# configure: error: yara not found
RUN apk add yara-dev
# configure: error: maxminddb not found
RUN apk add libmaxminddb-dev
# configure: error: please install magic library
RUN apk add file-dev
# configure: error: please install nghttp2 library
RUN apk add nghttp2-dev
# arkime.h:27:10: fatal error: glib.h: No such file or directory
RUN apk add glib-dev
# http.c:17:10: fatal error: curl/curl.h: No such file or director
RUN apk add curl-dev
# config.c:16:10: fatal error: yaml.h: No such file or directory
RUN apk add yaml-dev
# writer-simple.c:20:10: fatal error: zstd.h: No such file or directory
RUN apk add zstd-dev

# /usr/lib/gcc/x86_64-alpine-linux-musl/13.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reader-tpacketv3.o: in function `reader_tpacketv3_init':
# /src/capture/reader-tpacketv3.c:239:(.text+0x972): undefined reference to `mmap64'
RUN apk add patch
COPY <<-EOT mmap64.patch
---
 capture/reader-tpacketv3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/capture/reader-tpacketv3.c b/capture/reader-tpacketv3.c
index 6d4eacc..14a00be 100644
--- a/capture/reader-tpacketv3.c
+++ b/capture/reader-tpacketv3.c
@@ -235,7 +235,7 @@ void reader_tpacketv3_init(char *UNUSED(name))
                     CONFIGEXIT("Error setting SO_ATTACH_FILTER: %s", strerror(errno));
             }
 
-            infos[i][t].map = mmap64(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
+            infos[i][t].map = mmap(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
                                      PROT_READ | PROT_WRITE, MAP_SHARED | MAP_LOCKED, infos[i][t].fd, 0);
             if (unlikely(infos[i][t].map == MAP_FAILED)) {
                 CONFIGEXIT("MMap64 failure in reader_tpacketv3_init, %d: %s. Tried to allocate %d bytes (tpacketv3BlockSize: %d * 64) which was probbaly too large for this host, you probably need to reduce one of the values.", errno, strerror(errno), infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr, blocksize);
-- 
2.25.1
EOT

# Comment me in
# RUN patch --strip=1 --directory=/src --input=mmap64.patch

RUN ./configure

RUN make
```

Thanks,
H

--- a/capture/reader-tpacketv3.c
+++ b/capture/reader-tpacketv3.c
@@ -237,7 +237,7 @@ void reader_tpacketv3_init(char *UNUSED(name))
                     CONFIGEXIT("Error setting SO_ATTACH_FILTER: %s", strerror(errno));
             }
 
-            infos[i][t].map = mmap(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
+            infos[i][t].map = mmap(NULL, (size_t)infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
                                      PROT_READ | PROT_WRITE, MAP_SHARED | MAP_LOCKED, infos[i][t].fd, 0);
             if (unlikely(infos[i][t].map == MAP_FAILED)) {
                 CONFIGEXIT("MMap64 failure in reader_tpacketv3_init, %d: %s. Tried to allocate %d bytes (tpacketv3BlockSize: %d * 64) which was probbaly too large for this host, you probably need to reduce one of the values.", errno, strerror(errno), infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr, blocksize);

--- a/capture/reader-tpacketv3.c
+++ b/capture/reader-tpacketv3.c
@@ -13,6 +13,7 @@
  *
  */
 
+#define _FILE_OFFSET_BITS 64
 #include "arkime.h"
 extern ArkimeConfig_t        config;
 
@@ -236,7 +237,7 @@ void reader_tpacketv3_init(char *UNUSED(name))
                     CONFIGEXIT("Error setting SO_ATTACH_FILTER: %s", strerror(errno));
             }
 
-            infos[i][t].map = mmap64(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
+            infos[i][t].map = mmap(NULL, infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr,
                                      PROT_READ | PROT_WRITE, MAP_SHARED | MAP_LOCKED, infos[i][t].fd, 0);
             if (unlikely(infos[i][t].map == MAP_FAILED)) {
                 CONFIGEXIT("MMap64 failure in reader_tpacketv3_init, %d: %s. Tried to allocate %d bytes (tpacketv3BlockSize: %d * 64) which was probbaly too large for this host, you probably need to reduce one of the values.", errno, strerror(errno), infos[i][t].req.tp_block_size * infos[i][t].req.tp_block_nr, blocksize);

